### PR TITLE
Tests: Migrate from nodeunit to QUnit, uodate dependencies

### PR DESCRIPTION
**Pull request by mgol**

### CODE REVIEW

**Comment by mgol** [comment]
Since there are indentation changes, it might be a bit easier to review it with whitespace changes off.

File: test/package.js [hunk 1]
```diff
@@ -40,195 +42,204 @@ themeFiles = [
...
+
+		const done = assert.async();
+
+		const pkg = new Packager( context.files, Package, {
+			components: context.allComponents,
+			themeVars: defaultTheme
+		} );
+		pkg.toJson( function( error, files ) {
+			if ( error ) {
+				return done( error );
```
Comment (Krinkle) [inline test/package.js:68]: `done()` doesn't take any parameters. To reject tests with an error, you may want to promisify and return (or await) the promisified result instead.

**Review by Krinkle — COMMENTED** [review]
One nit, LGTM otherwise.

File: test/package.js (hunk 1 above)
Comment (mgol) [reply test/package.js]: @Krinkle Thanks for the catch! Updated.

**Review by timmywil — APPROVED** [review]
Nice!

**Review by Krinkle — APPROVED** [review]

**Comment by mgol** [comment]
Landed in 5f01a262e8fb0af8941df97307a68541873e5385.

**Pull request merged**